vivisect/symboliks/archs/i386.py

```diff
@@ -1373,6 +1393,33 @@ def i_shr(self, op):
         self.effSetVariable('eflags_sf', lt(res, zero))
         self.effSetVariable('eflags_eq', eq(res, zero))
 
+    def i_shrd(self, op):
+        dsize = op.opers[0].tsize
+        v1 = self.getOperObj(op, 0)
+        v2 = self.getOperObj(op, 1)
+        v3 = self.getOperObj(op, 2)
+        zero = Const(0, self._psize)
+
+        #res = (v1 | (v2 << Const(8*dsize, self._psize))) >> v3
+        res = (v1 >> v3) | (v2 << (Const(self._psize * 8, self._psize) - v3))
+
+        # No effect (not even flags) if shift is 0
+        if v3.isDiscrete():
+            bit = v3.solve()
+            if bit == 0:
+                return
+            elif bit == 1:
+                tsize = v1.getWidth()
+                power = o_pow(Const(2, tsize), Const(tsize - 1, self._psize), self._psize)
+                msb = o_div((v1 & power), power, v1.getWidth())
+                self.effSetVariable('eflags_of', msb)
```

Comment:
shld also has the portion about 1 bit shifts affect the overflow flag that we should add. (and technically a blurb about if the count is greater than the operand size, the flags being undefined)

Reply:
so... it seems like we need to have some if/then/else capability for the flags.  or some sort of constraint.  we often can't know until symbolik-emulation time what CL is going to be... so how do we best assert our will here?

Reply:
Hmm. Yea. We might want to discuss a new SymbolikEffect that collapses/calls a callback on apply/reduce. That might solve some of our branching logic states maybe?